store/sqlstore/post_store.go

```diff
@@ -734,21 +759,33 @@ type postIds struct {
 
 func (s *SqlPostStore) permanentDeleteAllCommentByUser(userId string) error {
 	results := []postIds{}
-	_, err := s.GetMaster().Select(&results, "Select Id, RootId FROM Posts WHERE UserId = :UserId AND RootId != ''", map[string]interface{}{"UserId": userId})
+	transaction, err := s.GetMaster().Begin()
+	if err != nil {
+		return errors.Wrap(err, "begin_transaction")
+	}
+	defer finalizeTransaction(transaction)
+
+	_, err = transaction.Select(&results, "Select Id, RootId FROM Posts WHERE UserId = :UserId AND RootId != ''", map[string]interface{}{"UserId": userId})
 	if err != nil {
 		return errors.Wrapf(err, "failed to fetch Posts with userId=%s", userId)
 	}
 
+	_, err = transaction.Exec("DELETE FROM Posts WHERE UserId = :UserId AND RootId != ''", map[string]interface{}{"UserId": userId})
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete Posts with userId=%s", userId)
+	}
+
 	for _, ids := range results {
-		if err = s.cleanupThreads(ids.Id, ids.RootId, true, userId); err != nil {
+		if err = s.cleanupThreads(transaction, ids.Id, ids.RootId, true, userId); err != nil {
```

Comment:
So now, it seems like we have a proper separation between `cleanupThreads` and `permanentDeleteThreads`. And in this case, it looks like we are always querying for `RootId != ''` which means the `permanent && rootId == ""` condition will always evaluate to false.

Wondering if we can totally remove the inner condition from `cleanupThreads` and rename it to `cleanupThreadComments`? That way we can get rid of the `permanent` parameter and always either call `cleanupThreadComments` or `permanentDeleteThreads`. Because this seems to be the only place where `cleanupThreads` is called with permanent as true.

Would that make sense?

Reply:
I've made those changes.
So to clarify on functionality.

- Deleting a post deletes the post and then cleans the threads if it's a comment, and does nothing for root posts. (there is another ticket about this)
- Permanent deletion of user's posts deletes all comments by user and cleans up those threads and then deletes all root posts by user and deletes threads and thread memberships for those root posts
- Permanent deletion of posts in channel, deletes all channel posts, comments and root posts, and then deletes all threads and thread memberships (no thread cleanup here, since we are deleting threads and thread memberships)

Reply:
Great stuff. Would be even better if you want to add this as a comment somewhere - maybe above `cleanupThreadComments` or `permanentDeleteThreads`. 0/5.